xcell/tests/test_cls.py

```diff
@@ -64,6 +64,19 @@ def test_smoke():
     get_cov_class()
 
 
+def test_cov_nlmarg():
+    data = get_config(0.2)
+    data['tracers']['Dummy__0']['nl_marginalize'] = True
+    data['tracers']['Dummy__0']['nl_prior'] = 1E30
+    data['tracers']['Dummy__0']['noise_level'] = 1E-5
+    cov_class = Cov(data, 'Dummy__0', 'Dummy__0', 'Dummy__0', 'Dummy__0')
+    cov = cov_class.get_covariance()
+    num_l = len(cov)
+    oo = np.ones(num_l)
+    chi2 = np.dot(oo, np.linalg.solve(cov, oo))
+    assert np.fabs(chi2) < 1E-5*num_l
```

Comment:
Why `<1 E-5 * num_l`?

Reply:
this is the chi2 with respect to a theory that is equivalent to the noise you just marginalized over, so it should be zero